lib/Providers/TaskProcessingProvider.php

```diff
@@ -61,10 +61,18 @@ public function getInputShapeEnumValues(): array {
 		$availableLanguages = $this->getAvailableLanguages();
 		$originLanguages = [];
 		$targetLanguages = [];
+		$originLanguageList = [];
+		$targetLanguageList = [];
 
 		foreach ($availableLanguages as $language) {
-			$originLanguages[] = new ShapeEnumValue($language->fromLabel, $language->from);
-			$targetLanguages[] = new ShapeEnumValue($language->toLabel, $language->to);
+			if (!in_array($language->fromLabel, $originLanguageList)) {
+				$originLanguages[] = new ShapeEnumValue($language->fromLabel, $language->from);
+				$originLanguageList[] = $language->fromLabel;
+			}
```

Comment:
let's use a dict instead of an array for performance benefits. Then use it like `array_values($originLanguages)` to extract the final array.
```suggestion
			if (!isset($originLanguages[$language->fromLabel])) {
				$originLanguages[$language->fromLabel] = new ShapeEnumValue($language->fromLabel, $language->from);
			}
```

Reply:
Sounds good, thank you!